checker-util/src/main/java/org/checkerframework/checker/regex/util/RegexUtil.java

```diff
@@ -362,57 +398,128 @@ private static int getGroupCount(Pattern p) {
     }
 
     /**
-     * Return the non-null groups in the argument.
+     * Returns a list of groups other than 0, that are guaranteed to be non-null given that the
+     * regular expression matches the String.
      *
-     * @param p pattern to be analysed
+     * @param regexp pattern to be analysed
```

Comment:
Is it required to be a valid regex, or might it be an invalid one?

Reply:
It has to be a regular expression. I realize only now that this is a public function. I added a `Pattern.compile(String)` to check for validity.

Reply:
My concern is with the documentation.  Callers need to know what arguments are legal, and they should not need to read the implementation.